### DISCUSSION TITLE
On save, everything is highlighted and formatted.

### DISCUSSION THREAD

**Discussion by JoshuaBlack0704** [helix-editor/helix#4022] [Q&A] [answered]
Every time I save after writing in the rust language helix will automatically highlight and format everything in the entire file. This is extremely annoying and I was wondering if there is anything in helix that might be causing this and what that might be so I can disable it?

**Comment by kirawi** [comment, marked as answer]
#2404

**Reply by JoshuaBlack0704** [reply]
Yep, format was the issue. Setting auto-format to false via "set auto-format false" fixed the issue. Although I'm not sure how it got turned on in the first place. Thank you for the speedy response.
